lib/appsignal/span.rb

```diff
@@ -0,0 +1,84 @@
+module Appsignal
+  class Span
+    def initialize(namespace = nil, ext = nil)
+      @ext = if ext
+               ext
+             else
+               Appsignal::Extension::Span.root(namespace || "")
+             end
+    end
+
+    def child
+      Span.new(nil, @ext.child)
+    end
+
+    def name=(value)
+      @ext.set_name(value)
+    end
+
+    def add_error(error)
+      unless error.is_a?(Exception)
+        Appsignal.logger.error "Appsignal::Span#add_error: Cannot add error. " \
+          "The given value is not an exception: #{error.inspect}"
+        return
+      end
+      return unless error
+
+      backtrace = cleaned_backtrace(error.backtrace)
+      @ext.add_error(
+        error.class.name,
+        error.message.to_s,
+        backtrace ? Appsignal::Utils::Data.generate(backtrace) : Appsignal::Extension.data_array_new
+      )
+    end
+
+    def set_sample_data(key, data)
+      return unless key && data && (data.is_a?(Array) || data.is_a?(Hash))
+      @ext.set_sample_data(
+        key.to_s,
+        Appsignal::Utils::Data.generate(data, true)
+      )
+    rescue RuntimeError => e
```

Comment:
Why RuntimeError specifically here? Why not StandardError (the Ruby default)?

Reply:
I cargoculted this from transaction, opened it up now.